interfaces/IMonitor.h

```diff
@@ -0,0 +1,80 @@
+ * limitations under the License.
+*/
+
+#pragma once
+
+#include "Module.h"
+
+// @stubgen:include <com/IIteratorType.h>
+
+namespace Thunder {
+namespace Exchange {
+    /* @json 1.0.0 */
+    struct EXTERNAL IMonitor : virtual public Core::IUnknown {
+        enum { ID = ID_MONITOR };
+        struct Measurement {
+            uint64_t min /* @brief Minimal value measured */;
+            uint64_t max /* @brief Maximal value measured */;
+            uint64_t average /* @brief Average of all measurements */;
+            uint64_t last /* @brief Last measured value */;
+        };
+        struct Measurements {
+            Measurement resident /* @brief Resident memory measurement */;
+            Measurement allocated /* @brief Allocated memory measurement */;
+            Measurement shared /* @brief Shared memory measurement */;
+            Measurement process /* @brief Processes measurement */;
+            bool operational /* @brief Whether the service is up and running */;
+            uint32_t count /* @brief Number of measurements */;
+        };
+        struct RestartInfo {
+            uint8_t limit /* @brief Maximum number or restarts to be attempted */;
+            uint16_t window /* @brief Time period (in seconds) within which failures must happen for the limit to be considered crossed */;
+        };
+        struct Statistics {
+            Measurements measurements /* @brief Measurements for the service */;
+            string observable /* @brief A callsign of the watched service */;
+            RestartInfo restart /* @brief Restart limits for failures applying to the service */;
+        };
+        struct ActionParams {
+            string callsign /* @brief Callsign of the service the Monitor acted upon */;
+            string action /* @brief The action executed by the Monitor on a service. One of: "Activate", "Deactivate", "StoppedRestarting" */;
+            string reason /* @brief A message describing the reason the action was taken */;
+        };
+        struct RestartLimitsInfo {
+            string callsign /* @brief The callsign of a service to reset measurements snapshot of */;
+            RestartInfo restart /* @brief Restart limits for failures applying to the service */;
+        };
+
+        struct EXTERNAL INotification : virtual public Core::IUnknown {
+            enum { ID = ID_MONITOR_NOTIFICATION };
+            virtual void Action(const string& callsign, const string& action, const string& reason) = 0;
+        };
+
+        using IStatisticsIterator = RPC::IIteratorType<Statistics, ID_MONITOR_STATISTICS_ITERATOR>;
+        // @brief RestartLimits: Sets new restart limits for a plugin
+        virtual uint32_t RestartLimits(const RestartLimitsInfo& params) = 0;
+        // @brief ResetStats: Resets memory and process statistics for a single plugin watched by the Monitor
+        virtual uint32_t ResetStats(const string& callsign, Statistics& statistics /* @out */) = 0;
+        // @property
```

Comment:
It does not reset the reastartInformation, only the measurements, so suggest:
```
virtual Core::hresult Reset(const string& callsign, Measurements& currentMeasurements /* @out */) = 0;

Reply:
It will do the reset, so I changed the interface to support only reset : virtual uint32_t Reset(const string& callsign) = 0;, the status can be retrieve with other interface Status()